server/main_server/src/rodsServer.cpp

```diff
@@ -1025,11 +1025,11 @@ int main(int argc, char** argv)
 
                 try {
                     // Collapse non-zero error codes into one.
-                    _exit(runIrodsAgentFactory(local_addr) == 0 ? 0 : 1);
+                    exit(runIrodsAgentFactory(local_addr) == 0 ? 0 : 1);
                 }
                 catch (...) {
                     // If an exception is thrown for any reason, terminate the agent factory.
-                    _exit(1);
+                    exit(1);
```

Comment:
Consider whether this is correct.

Running this in the test suite should help answer this question since the introduction of `_exit` was driven by docker.

Reply:
Leave a comment about `runIrodsAgentFactory` never returning on shutdown.

Reply:
The agent factory is normally shut down via SIGTERM. However, if the agent factory fails to fork an agent, it will return SYS_FORK_ERROR up the call chain.

If the agent factory crashes for whatever reason, respawned by the main server process, and then fails to fork an agent, the agent factory will most likely exit with a SIGABRT due to a failed assertion in boost::mutex or boost::condition_variable.

The only way around this is to continue using `_exit` in the cron manager task for the agent factory.

Basically the change captured here needs to be reverted to its original form.

I'll add this as the comment to the code.

Reply:
Done.